kafka-rest/src/main/java/io/confluent/kafkarest/controllers/ConsumerLagManagerImpl.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.controllers;
+
+import static java.util.Objects.requireNonNull;
+
+import io.confluent.kafkarest.common.KafkaFutures;
+import io.confluent.kafkarest.entities.ConsumerLag;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.ListConsumerGroupOffsetsOptions;
+import org.apache.kafka.clients.admin.ListOffsetsOptions;
+import org.apache.kafka.clients.admin.OffsetSpec;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.IsolationLevel;
+import org.apache.kafka.common.TopicPartition;
+
+final class ConsumerLagManagerImpl implements ConsumerLagManager {
+
+  private final Admin adminClient;
+  private final ConsumerOffsetsDao consumerOffsetsDao;
+  private final IsolationLevel isolationLevel = IsolationLevel.READ_COMMITTED;
+
+  @Inject
+  ConsumerLagManagerImpl(Admin adminClient, ConsumerOffsetsDao consumerOffsetsDao) {
+    this.adminClient = requireNonNull(adminClient);
+    this.consumerOffsetsDao = requireNonNull(consumerOffsetsDao);
+  }
+
+  @Override
+  public CompletableFuture<List<ConsumerLag>> listConsumerLags(
+      String clusterId, String consumerGroupId) {
+    try {
+      return CompletableFuture.completedFuture(
+          consumerOffsetsDao.getConsumerLags(
```

Comment:
It looks like we have some async apis and some sync apis. Any reason they are not all async?

Reply:
Making them all async!